### ISSUE TITLE
Multiline named imports

### ISSUE THREAD

**Issue by jimafisk** [plentico/plenti#175]
While working on adding the git-cms, I noticed that named imports don't always work. For example, the following breaks the build:

```js
import { requestAuthCode, requestAccessToken, requestRefreshToken } from './cms/auth.svelte';
```

Error message: `javascript stack trace: SyntaxError: Cannot use import statement outside a module`

</details>

But if you reorganize it differently, it works fine:
```js
import { requestAuthCode, requestAccessToken } from './cms/auth.svelte';
import { requestRefreshToken } from './cms/auth.svelte';
```

The problem appears to be that during the build if this gets too long it gets converted to a multiline import:

<details>
<summary>See screenshot</summary>

![multiline](https://user-images.githubusercontent.com/5913244/149413285-92a15479-477e-47d1-b600-5aedc8dd1354.png)

</details>

This seems to be happening somewhere around: https://github.com/plentico/plenti/blob/0a14d97a01182f9a3f5948a872e58196c7cb862c/cmd/build/compile.go#L173

**Comment by jimafisk** [comment]
Tried fixing like:

```go
// Convert multiline imports into one line so they can be processed.
singleLineNamedImport := strings.Replace(namedImport, "\n", "", -1)
// Remove extra space around named imports
singleLineNamedImport = strings.Join(strings.Fields(strings.TrimSpace(singleLineNamedImport)), " ")
```

Which seems to have correctly converted multiline into a single line: 

<details>
<summary>Previous output</summary>

```
[import {
        requestAuthCode,
        requestAccessToken,
        requestRefreshToken
} from "./cms/auth.svelte";   {
        requestAuthCode,
        requestAccessToken,
        requestRefreshToken
} from "./cms/auth.svelte";  {
        requestAuthCode,
        requestAccessToken,
        requestRefreshToken
        requestRefreshToken
    "./cms/auth.svelte"]
```

</details>

<details>
<summary>Output with newlines removed but not spaces</summary>

```
[import {       requestAuthCode,        requestAccessToken,     requestRefreshToken} from {  requestAuthCode,        requestAccessToken,     requestRefreshToken}]
```

</details>
<details>
<summary>Output with newlines and spaces removed</summary>

```
[import { requestAuthCode, requestAccessToken, requestRefreshToken} from { requestAuthCode, requestAccessToken, requestRefreshToken}]
```

</details>

However the problem still persists. It looks like all multiline named imports are getting assigned to the `'../web_modules/svelte/internal/index.mjs'` path. This is normal for things like Svelte helper functions:

<details>
<summary>Svelte helper functions</summary>

```js
import {
	SvelteComponent,
	claim_component,
	create_component,
	destroy_component,
	init,
	mount_component,
	safe_not_equal,
	transition_in,
	transition_out
} from '../web_modules/svelte/internal/index.mjs';
```

</details>

But we don't want this for our custom named imports:

<details>
<summary>Wrong path for custom named imports</summary>

```js
import {
	requestAuthCode,
	requestAccessToken,
	requestRefreshToken
} from '../web_modules/svelte/internal/index.mjs';
```

</details>

**Comment by jimafisk** [comment]
Turns out the fix was needed in `cmd/build/gopack.go`. The regexp used to find the import paths was greedy:

```go
reStaticImportGoPk = regexp.MustCompile(`(?m)^import(\s)(.*from(.*);|((.*\n){0,})\}(\s)from(.*);)`)
```
<details>
<summary>Matches multiple imports if more than one multiline import exists in a file</summary>

```
FoundPath = ../web_modules/svelte/internal/index.mjs
import {
        SvelteComponent,
        claim_component,
        create_component,
        destroy_component,
        init,
        mount_component,
        safe_not_equal,
        transition_in,
        transition_out
} from "svelte/internal";

import Navaid from "navaid";
import Html from "../global/html.svelte";
import { getContent } from "./main.js";

// Start CMS
import {
        requestAuthCode,
        requestAccessToken,
        requestRefreshToken
} from "./cms/auth.svelte";
```
</details>

I updated it to use a lazy quantifier (note the added question mark to `{0,}?`):

```go
reStaticImportGoPk = regexp.MustCompile(`(?m)^import(\s)(.*from(.*);|((.*\n){0,}?)\}(\s)from(.*);)`)
```

<details>
<summary>Matches only the current multiline import</summary>

```
FoundPath = ../web_modules/svelte/internal/index.mjs
import {
        SvelteComponent,
        claim_component,
        create_component,
        destroy_component,
        init,
        mount_component,
        safe_not_equal,
        transition_in,
        transition_out
} from "svelte/internal";
```

</details>

**Comment by jimafisk** [comment]
Noticed some issues with this detailed here: https://github.com/plentico/plenti/pull/176#issuecomment-1013798448

Using the `plenti-tags-example` for testing and debugging with:

```go
// Get all the import statements.
staticImportStatements := reStaticImportGoPk.FindAll(contentBytes, -1)
for _, s := range staticImportStatements {
	fmt.Println(string(s))
}
```

<details>
<summary>Previously</summary>

```
import Router from './router.svelte';
import allContent from './content.js';
import * as allLayouts from './layouts.js';
import { env } from './env.js';
import {
	SvelteComponent,
	claim_component,
	create_component,
	destroy_component,
	init,
	mount_component,
	safe_not_equal,
	transition_in,
	transition_out
} from "svelte/internal";

import Navaid from "navaid";
import Html from "../global/html.svelte";
import { getContent } from "./main.js";

// Git-CMS
import {
	requestAuthCode,
	requestAccessToken,
	requestRefreshToken
} from "./cms/auth.js";
import { session } from "./cms/session.js";
import { storage } from "./cms/storage.js";
import { onMount } from "svelte";
import AdminMenu from "./cms/admin_menu.svelte";
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	detach,
	element,
	init,
	insert,
	listen,
	noop,
	prevent_default,
	safe_not_equal,
	set_style,
	space,
	svg_element
} from "svelte/internal";
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	claim_text,
	destroy_each,
	detach,
	element,
	empty,
	init,
	insert,
	noop,
	safe_not_equal,
	set_data,
	space,
	text
} from "svelte/internal";
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	claim_text,
	destroy_each,
	detach,
	element,
	init,
	insert,
	noop,
	safe_not_equal,
	set_data,
	space,
	text
} from "svelte/internal";
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	claim_text,
	destroy_each,
	detach,
	element,
	empty,
	init,
	insert,
	noop,
	safe_not_equal,
	set_data,
	space,
	text
} from "svelte/internal";
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	claim_text,
	detach,
	element,
	init,
	insert,
	noop,
	safe_not_equal,
	set_data,
	space,
	text
} from "svelte/internal";
import {
	SvelteComponent,
	append,
	assign,
	attr,
	check_outros,
	children,
	claim_component,
	claim_element,
	claim_space,
	create_component,
	destroy_component,
	detach,
	element,
	get_spread_object,
	get_spread_update,
	group_outros,
	init,
	insert,
	mount_component,
	safe_not_equal,
	space,
	transition_in,
	transition_out
} from "svelte/internal";
import Head from "./head.svelte";
```

</details>

<details>
<summary>Now</summary>

```
import Router from './router.svelte';
import allContent from './content.js';
import * as allLayouts from './layouts.js';
import { env } from './env.js';
import {
	SvelteComponent,
	claim_component,
	create_component,
	destroy_component,
	init,
	mount_component,
	safe_not_equal,
	transition_in,
	transition_out
} from "svelte/internal";
import Navaid from "navaid";
import Html from "../global/html.svelte";
import { getContent } from "./main.js";
import {
	requestAuthCode,
	requestAccessToken,
	requestRefreshToken
} from "./cms/auth.js";
import { session } from "./cms/session.js";
import { storage } from "./cms/storage.js";
import { onMount } from "svelte";
import AdminMenu from "./cms/admin_menu.svelte";
import convert from 'regexparam';
import {
	SvelteComponent,
	append,
	assign,
	attr,
	check_outros,
	children,
	claim_component,
	claim_element,
	claim_space,
	create_component,
	destroy_component,
	detach,
	element,
	get_spread_object,
	get_spread_update,
	group_outros,
	init,
	insert,
	mount_component,
	safe_not_equal,
	space,
	transition_in,
	transition_out
} from "svelte/internal";
import Head from "./head.svelte";
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	claim_text,
	detach,
	element,
	init,
	insert,
	noop,
	safe_not_equal,
	set_data,
	space,
	text
} from "svelte/internal";
import Router from './router.svelte';
import allContent from './content.js';
import * as allLayouts from './layouts.js';
import { env } from './env.js';
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	claim_text,
	destroy_each,
	detach,
	element,
	empty,
	init,
	insert,
	noop,
	safe_not_equal,
	set_data,
	space,
	text
} from "svelte/internal";
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	claim_text,
	destroy_each,
	detach,
	element,
	init,
	insert,
	noop,
	safe_not_equal,
	set_data,
	space,
	text
} from "svelte/internal";
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	claim_text,
	destroy_each,
	detach,
	element,
	empty,
	init,
	insert,
	noop,
	safe_not_equal,
	set_data,
	space,
	text
} from "svelte/internal";
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	claim_text,
	detach,
	element,
	init,
	insert,
	noop,
	safe_not_equal,
	set_data,
	space,
	text
} from '../web_modules/svelte/internal/index.mjs';
import { session } from './session.js';
import { storage } from './storage.js';
import { env } from '../env.js';
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	detach,
	element,
	init,
	insert,
	listen,
	noop,
	prevent_default,
	safe_not_equal,
	set_style,
	space,
	svg_element
} from "svelte/internal";
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	claim_text,
	destroy_each,
	detach,
	element,
	empty,
	init,
	insert,
	noop,
	safe_not_equal,
	set_data,
	space,
	text
} from '../web_modules/svelte/internal/index.mjs';
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	claim_text,
	destroy_each,
	detach,
	element,
	init,
	insert,
	noop,
	safe_not_equal,
	set_data,
	space,
	text
} from '../web_modules/svelte/internal/index.mjs';
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	claim_text,
	destroy_each,
	detach,
	element,
	empty,
	init,
	insert,
	noop,
	safe_not_equal,
	set_data,
	space,
	text
} from '../web_modules/svelte/internal/index.mjs';
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	claim_text,
	detach,
	element,
	init,
	insert,
	noop,
	safe_not_equal,
	set_data,
	space,
	text
} from '../web_modules/svelte/internal/index.mjs';
import {
	SvelteComponent,
	append,
	assign,
	attr,
	check_outros,
	children,
	claim_component,
	claim_element,
	claim_space,
	create_component,
	destroy_component,
	detach,
	element,
	get_spread_object,
	get_spread_update,
	group_outros,
	init,
	insert,
	mount_component,
	safe_not_equal,
	space,
	transition_in,
	transition_out
} from '../web_modules/svelte/internal/index.mjs';
import Head from './head.js';
```

</details>

**Comment by jimafisk** [comment]
Similarly with:
```go
// Get all the exported import statements.
staticExportStatements := reStaticExportGoPk.FindAll(contentBytes, -1)
for _, s := range staticExportStatements {
	fmt.Println(string(s))
}
```

<details>
<summary>Before</summary>

```
export { SvelteComponentDev as SvelteComponent, afterUpdate, beforeUpdate, createEventDispatcher, getContext, onDestroy, onMount, setContext, tick } from './internal';
export {default as layouts_content_index_svelte} from '../content/index.svelte';
export {default as layouts_content_posts_svelte} from '../content/posts.svelte';
export {default as layouts_content_tags_svelte} from '../content/tags.svelte';
export {default as layouts_global_head_svelte} from '../global/head.svelte';
export {default as layouts_global_html_svelte} from '../global/html.svelte';
```

</details>

<details>
<summary>Now</summary>

```
export {default as layouts_content_index_svelte} from '../content/index.svelte';
export {default as layouts_content_posts_svelte} from '../content/posts.svelte';
export {default as layouts_content_tags_svelte} from '../content/tags.svelte';
export {default as layouts_global_head_svelte} from '../global/head.svelte';
export {default as layouts_global_html_svelte} from '../global/html.svelte';
export { SvelteComponentDev as SvelteComponent, afterUpdate, beforeUpdate, createEventDispatcher, getContext, onDestroy, onMount, setContext, tick } from './internal';
export {default as layouts_content_index_svelte} from '../content/index.js';
export {default as layouts_content_posts_svelte} from '../content/posts.js';
export {default as layouts_content_tags_svelte} from '../content/tags.js';
export {default as layouts_global_head_svelte} from '../global/head.js';
export {default as layouts_global_html_svelte} from '../global/html.js';
```

</details>

**Comment by jimafisk** [comment]
What I think might be happening here: Since we're now effectively getting individual import statements instead of sometimes accidentally grouping them, we're checking more paths than we used to. Sometimes the path will be the same as another import statement, but we add those to an `alreadyConvertedFiles` array so we don't have to rerun them. We might be checking for all the import statements, not finding any in the case of a file like `/spa/web_modules/svelte/index.mjs ` (uses [directly exported import statements](https://stackoverflow.com/questions/38077164/es6-export-from-import)) and adding it to the `alreadyConvertedFiles` before we can check if there is an exported import statement on that file.

**Comment by jimafisk** [comment]
With the lazy quantifier `?` in the regexp, we seem to be getting lots of duplicates:

<details>
<summary>Every fullPathStr inside !alreadyConverted check</summary>

```
public/spa/ejected/router.js
public/spa/web_modules/svelte/internal/index.mjs
public/spa/web_modules/navaid/dist/navaid.mjs
public/spa/web_modules/regexparam/dist/regexparam.mjs
public/spa/global/html.js
public/spa/global/head.js
public/spa/ejected/main.js
public/spa/ejected/content.js
public/spa/ejected/layouts.js
public/spa/content/index.js
public/spa/content/posts.js
public/spa/content/tags.js
public/spa/global/head.js
public/spa/ejected/env.js
public/spa/ejected/cms/auth.js
public/spa/ejected/cms/session.js
public/spa/ejected/cms/storage.js
public/spa/ejected/env.js
public/spa/ejected/cms/session.js
public/spa/ejected/cms/storage.js
public/spa/web_modules/svelte/index.mjs
public/spa/web_modules/svelte/internal
public/spa/ejected/cms/admin_menu.js
public/spa/ejected/content.js
public/spa/ejected/layouts.js
public/spa/content/index.js
public/spa/web_modules/svelte/internal/index.mjs
public/spa/content/posts.js
public/spa/web_modules/svelte/internal/index.mjs
public/spa/content/tags.js
public/spa/web_modules/svelte/internal/index.mjs
public/spa/global/head.js
public/spa/web_modules/svelte/internal/index.mjs
public/spa/global/html.js
public/spa/web_modules/svelte/internal/index.mjs
public/spa/ejected/env.js
```

</details>

<details>
<summary>convertPath and foundPath inside foundPath != ""</summary>

```
CONVERTPATH: public/spa/ejected/router.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs
CONVERTPATH: public/spa/web_modules/navaid/dist/navaid.mjs
FOUNDPATH: ../../regexparam/dist/regexparam.mjs
CONVERTPATH: public/spa/ejected/router.js
FOUNDPATH: ../web_modules/navaid/dist/navaid.mjs
CONVERTPATH: public/spa/global/html.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs
CONVERTPATH: public/spa/global/head.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs
CONVERTPATH: public/spa/global/html.js
FOUNDPATH: ./head.js
CONVERTPATH: public/spa/ejected/router.js
FOUNDPATH: ../global/html.js
CONVERTPATH: public/spa/ejected/main.js
FOUNDPATH: ./router.js
CONVERTPATH: public/spa/ejected/main.js
FOUNDPATH: ./content.js
CONVERTPATH: public/spa/content/index.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs
CONVERTPATH: public/spa/ejected/layouts.js
FOUNDPATH: ../content/index.js
CONVERTPATH: public/spa/content/posts.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs
CONVERTPATH: public/spa/ejected/layouts.js
FOUNDPATH: ../content/posts.js
CONVERTPATH: public/spa/content/tags.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs
CONVERTPATH: public/spa/ejected/layouts.js
FOUNDPATH: ../content/tags.js
CONVERTPATH: public/spa/global/head.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs
CONVERTPATH: public/spa/ejected/layouts.js
FOUNDPATH: ../global/head.js
CONVERTPATH: public/spa/ejected/layouts.js
FOUNDPATH: ../global/html.js
CONVERTPATH: public/spa/ejected/main.js
FOUNDPATH: ./layouts.js
CONVERTPATH: public/spa/ejected/main.js
FOUNDPATH: ./env.js
CONVERTPATH: public/spa/ejected/router.js
FOUNDPATH: ./main.js
CONVERTPATH: public/spa/ejected/cms/auth.js
FOUNDPATH: ./session.js
CONVERTPATH: public/spa/ejected/cms/auth.js
FOUNDPATH: ./storage.js
CONVERTPATH: public/spa/ejected/cms/auth.js
FOUNDPATH: ../env.js
CONVERTPATH: public/spa/ejected/router.js
FOUNDPATH: ./cms/auth.js
CONVERTPATH: public/spa/ejected/router.js
FOUNDPATH: ./cms/session.js
CONVERTPATH: public/spa/ejected/router.js
FOUNDPATH: ./cms/storage.js
CONVERTPATH: public/spa/web_modules/svelte/index.mjs
FOUNDPATH: ./internal
CONVERTPATH: public/spa/ejected/router.js
FOUNDPATH: ../web_modules/svelte/index.mjs
CONVERTPATH: public/spa/ejected/cms/admin_menu.js
FOUNDPATH: ../../web_modules/svelte/internal/index.mjs
CONVERTPATH: public/spa/ejected/router.js
FOUNDPATH: ./cms/admin_menu.js
CONVERTPATH: public/spa/ejected/main.js
FOUNDPATH: ./router.js
CONVERTPATH: public/spa/ejected/main.js
FOUNDPATH: ./content.js
CONVERTPATH: public/spa/content/index.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs
CONVERTPATH: public/spa/ejected/layouts.js
FOUNDPATH: ../content/index.js
CONVERTPATH: public/spa/content/posts.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs
CONVERTPATH: public/spa/ejected/layouts.js
FOUNDPATH: ../content/posts.js
CONVERTPATH: public/spa/content/tags.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs
CONVERTPATH: public/spa/ejected/layouts.js
FOUNDPATH: ../content/tags.js
CONVERTPATH: public/spa/global/head.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs
CONVERTPATH: public/spa/ejected/layouts.js
FOUNDPATH: ../global/head.js
CONVERTPATH: public/spa/global/html.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs
CONVERTPATH: public/spa/global/html.js
FOUNDPATH: ./head.js
CONVERTPATH: public/spa/ejected/layouts.js
FOUNDPATH: ../global/html.js
CONVERTPATH: public/spa/ejected/main.js
FOUNDPATH: ./layouts.js
CONVERTPATH: public/spa/ejected/main.js
FOUNDPATH: ./env.js
```

</details>

<details>
<summary>Copied the data above and grouped by convertPath</summary>

```
CONVERTPATH: public/spa/ejected/router.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs
FOUNDPATH: ../web_modules/navaid/dist/navaid.mjs
FOUNDPATH: ../global/html.js
FOUNDPATH: ./main.js
FOUNDPATH: ./cms/auth.js
FOUNDPATH: ./cms/session.js
FOUNDPATH: ./cms/storage.js
FOUNDPATH: ../web_modules/svelte/index.mjs
FOUNDPATH: ./cms/admin_menu.js

CONVERTPATH: public/spa/web_modules/navaid/dist/navaid.mjs
FOUNDPATH: ../../regexparam/dist/regexparam.mjs

CONVERTPATH: public/spa/global/html.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs
FOUNDPATH: ./head.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs
FOUNDPATH: ./head.js

CONVERTPATH: public/spa/global/head.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs
FOUNDPATH: ../web_modules/svelte/internal/index.mjs

CONVERTPATH: public/spa/ejected/main.js
FOUNDPATH: ./router.js
FOUNDPATH: ./content.js

CONVERTPATH: public/spa/content/index.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs
FOUNDPATH: ../web_modules/svelte/internal/index.mjs

CONVERTPATH: public/spa/ejected/layouts.js
FOUNDPATH: ../content/index.js
FOUNDPATH: ../content/posts.js
FOUNDPATH: ../content/tags.js
FOUNDPATH: ../global/head.js
FOUNDPATH: ../global/html.js
FOUNDPATH: ../global/html.js
FOUNDPATH: ../content/posts.js
FOUNDPATH: ../content/index.js
FOUNDPATH: ../content/tags.js
FOUNDPATH: ../global/head.js

CONVERTPATH: public/spa/content/posts.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs
FOUNDPATH: ../web_modules/svelte/internal/index.mjs

CONVERTPATH: public/spa/content/tags.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs
FOUNDPATH: ../web_modules/svelte/internal/index.mjs

CONVERTPATH: public/spa/global/head.js
FOUNDPATH: ../web_modules/svelte/internal/index.mjs

CONVERTPATH: public/spa/ejected/main.js
FOUNDPATH: ./layouts.js
FOUNDPATH: ./env.js
FOUNDPATH: ./router.js
FOUNDPATH: ./content.js
FOUNDPATH: ./layouts.js
FOUNDPATH: ./env.js

CONVERTPATH: public/spa/ejected/cms/auth.js
FOUNDPATH: ./session.js
FOUNDPATH: ./storage.js
FOUNDPATH: ../env.js

CONVERTPATH: public/spa/web_modules/svelte/index.mjs
FOUNDPATH: ./internal

CONVERTPATH: public/spa/ejected/cms/admin_menu.js
FOUNDPATH: ../../web_modules/svelte/internal/index.mjs
```

</details>

**Comment by jimafisk** [comment]
I'm trying to figure out the exact order of when each path is converted and which parent file is referencing that path. Here's some prints from the [plenti-tags-example](https://github.com/jimafisk/plenti-tags-example) project:

<details>
<summary>Debugging output</summary>

```
FROM: public/spa/ejected/main.js
import('../content/' + content.type + '.js')
DYNAMIC IMPORT ^
import Router from './router.svelte';
STATIC IMPORT ^
import allContent from './content.js';
STATIC IMPORT ^
import * as allLayouts from './layouts.js';
STATIC IMPORT ^
import { env } from './env.js';
STATIC IMPORT ^



FROM: public/spa/ejected/router.js
import("../content/404.js")
DYNAMIC IMPORT ^
import("../content/" + content.type + ".js")
DYNAMIC IMPORT ^
import {
	SvelteComponent,
	claim_component,
	create_component,
	destroy_component,
	init,
	mount_component,
	safe_not_equal,
	transition_in,
	transition_out
} from "svelte/internal";
STATIC IMPORT ^
import Navaid from "navaid";
STATIC IMPORT ^
import Html from "../global/html.svelte";
STATIC IMPORT ^
import { getContent } from "./main.js";
STATIC IMPORT ^
import {
	requestAuthCode,
	requestAccessToken,
	requestRefreshToken
} from "./cms/auth.js";
STATIC IMPORT ^
import { session } from "./cms/session.js";
STATIC IMPORT ^
import { storage } from "./cms/storage.js";
STATIC IMPORT ^
import { onMount } from "svelte";
STATIC IMPORT ^
import AdminMenu from "./cms/admin_menu.svelte";
STATIC IMPORT ^



FROM: public/spa/web_modules/svelte/internal/index.mjs



FROM: public/spa/web_modules/navaid/dist/navaid.mjs
import convert from 'regexparam';
STATIC IMPORT ^



FROM: public/spa/web_modules/regexparam/dist/regexparam.mjs



FROM: public/spa/global/html.js
import {
	SvelteComponent,
	append,
	assign,
	attr,
	check_outros,
	children,
	claim_component,
	claim_element,
	claim_space,
	create_component,
	destroy_component,
	detach,
	element,
	get_spread_object,
	get_spread_update,
	group_outros,
	init,
	insert,
	mount_component,
	safe_not_equal,
	space,
	transition_in,
	transition_out
} from "svelte/internal";
STATIC IMPORT ^
import Head from "./head.svelte";
STATIC IMPORT ^



FROM: public/spa/global/head.js
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	claim_text,
	detach,
	element,
	init,
	insert,
	noop,
	safe_not_equal,
	set_data,
	space,
	text
} from "svelte/internal";
STATIC IMPORT ^



FROM: public/spa/ejected/main.js
import('../content/' + content.type + '.js')
DYNAMIC IMPORT ^
import Router from './router.svelte';
STATIC IMPORT ^
import allContent from './content.js';
STATIC IMPORT ^
import * as allLayouts from './layouts.js';
STATIC IMPORT ^
import { env } from './env.js';
STATIC IMPORT ^



FROM: public/spa/ejected/content.js



FROM: public/spa/ejected/layouts.js
export {default as layouts_content_index_svelte} from '../content/index.svelte';
STATIC EXPORT ^
export {default as layouts_content_posts_svelte} from '../content/posts.svelte';
STATIC EXPORT ^
export {default as layouts_content_tags_svelte} from '../content/tags.svelte';
STATIC EXPORT ^
export {default as layouts_global_head_svelte} from '../global/head.svelte';
STATIC EXPORT ^
export {default as layouts_global_html_svelte} from '../global/html.svelte';
STATIC EXPORT ^



FROM: public/spa/content/index.js
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	claim_text,
	destroy_each,
	detach,
	element,
	empty,
	init,
	insert,
	noop,
	safe_not_equal,
	set_data,
	space,
	text
} from "svelte/internal";
STATIC IMPORT ^



FROM: public/spa/content/posts.js
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	claim_text,
	destroy_each,
	detach,
	element,
	init,
	insert,
	noop,
	safe_not_equal,
	set_data,
	space,
	text
} from "svelte/internal";
STATIC IMPORT ^



FROM: public/spa/content/tags.js
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	claim_text,
	destroy_each,
	detach,
	element,
	empty,
	init,
	insert,
	noop,
	safe_not_equal,
	set_data,
	space,
	text
} from "svelte/internal";
STATIC IMPORT ^



FROM: public/spa/ejected/env.js



FROM: public/spa/ejected/cms/auth.js
import { session } from './session.js';
STATIC IMPORT ^
import { storage } from './storage.js';
STATIC IMPORT ^
import { env } from '../env.js';
STATIC IMPORT ^



FROM: public/spa/ejected/cms/session.js



FROM: public/spa/ejected/cms/storage.js



FROM: public/spa/web_modules/svelte/index.mjs
export { SvelteComponentDev as SvelteComponent, afterUpdate, beforeUpdate, createEventDispatcher, getContext, onDestroy, onMount, setContext, tick } from './internal';
STATIC EXPORT ^



FROM: public/spa/ejected/cms/admin_menu.js
import {
	SvelteComponent,
	append,
	attr,
	children,
	claim_element,
	claim_space,
	detach,
	element,
	init,
	insert,
	listen,
	noop,
	prevent_default,
	safe_not_equal,
	set_style,
	space,
	svg_element
} from "svelte/internal";
STATIC IMPORT ^
```

</details>

I've mapped out a visualization of the order in which the paths are converted given the recursive nature of how gopack finds paths from imports and follows those to find other paths:

<details>
<summary>Visualization</summary>

![IMG_20220118_141831509](https://user-images.githubusercontent.com/5913244/150004323-53f38645-107e-42a8-a072-9d0de457571c.jpg)


</details>

**Comment by jimafisk** [comment]
| level 1 | level 2 | level 3 | level 4 | Notes |
|----------|-----------|----------|----------|-----------|
| spa/ejected/main.js | | | | |
| | main --> spa/ejected/router.js | | | |
| | | router--> spa/web_modules/svelte/internal/index.mjs | | |
| | | router --> spa/web_modules/navaid/dist/navaid.mjs | | |
| | | | navaid --> spa/web_modules/regexparam/dist/regexparam.mjs | |
| | | router --> spa/global/html.js | | |
| | | | html --> spa/global/head.js | |
| router --> spa/ejected/main.js | | router --> spa/ejected/main.js | | (circular reference) |
| | main --> spa/ejected/content.js | | | |
| | main --> spa/ejected/layouts.js | | | |
| | | layouts --> spa/content/index.js | | |
| | | layouts --> spa/content/posts.js | | |
| | | layouts --> spa/content/tags.js | | |
| | main --> spa/ejected/env.js | | | |
| | | router --> spa/ejected/cms/auth.js | | |
| | | | auth --> spa/ejected/cms/session.js | |
| | | | auth --> spa/ejected/cms/storage.js | |
| | | router (onMount) --> spa/web_modules/svelte/index.mjs | | |
| | | router --> spa/ejected/cms/admin_menu.js | | |

**Comment by jimafisk** [comment]
It turns out the problem was the import in http://localhost:3000/spa/web_modules/svelte/index.mjs:
```js
export { SvelteComponentDev as SvelteComponent, afterUpdate, beforeUpdate, createEventDispatcher, getContext, onDestroy, onMount, setContext, tick } from './internal';
```

Was referencing a folder `./internal` instead of the specific file like `./internal/index.mjs`. I assume they do this to allow commonjs and esm support so your tooling can pick whichever one will work with your setup. I needed to modified Gopack again, we hadn't account for relative imports in an NPM that need to lookup nested `package.json` info to determine how to resolve the path. Should be fixed in https://github.com/plentico/plenti/pull/176/commits/1f36fb649ba6814bcbd7163050cff9a4d0e01a0a

**Comment by jimafisk** [comment]
Should be fixed in [v0.5.0](https://github.com/plentico/plenti/releases/tag/v0.5.0)
